hapi-fhir-jpaserver-model/src/main/java/ca/uhn/fhir/jpa/model/entity/ResourceIndexedSearchParamToken.java

```diff
@@ -370,4 +371,10 @@ public BaseResourceIndexedSearchParam setResource(ResourceTable theResource) {
 		setResourceType(theResource.getResourceType());
 		return this;
 	}
+
+	@PrePersist
+	public void truncateValueTo200Chars() {
```

Comment:
This is weird name -- just use max.
Also: magic like this needs some documentation
```suggestion
/**
 * We truncate the fields at the last moment because the tables have limited size.
 * We don't truncate earlier in the flow because the index hashes MUST be calculated on the full string.
 */
	public void truncateFieldsForDB() {
```